core/src/main/java/fr/sncf/osrd/speedcontroller/generators/ConstructionAllowanceGenerator.java

```diff
@@ -56,59 +54,43 @@ protected double getFirstGuess() {
-            SpeedController brakingSpeedController;
-            if (schedule.rollingStock.gammaType == RollingStock.GammaType.CONST) {
-                brakingSpeedController = new LimitAnnounceSpeedController(
-                        initialSpeed * scaleFactor,
-                        initialPosition,
-                        endBrakingPosition,
-                        schedule.rollingStock.gamma);
-            } else {
-                //TODO: optimise, this calculation is done twice
-                var updatesMap = getStepsAtPositionsToTarget(sim, schedule,
-                        initialPosition, initialSpeed, endBrakingPosition, initialSpeed * scaleFactor);
-                var speeds = new SortedDoubleMap();
-                for (var k : updatesMap.keySet()) {
-                    speeds.put(k, updatesMap.get(k).finalSpeed);
-                }
-                brakingSpeedController = new BrakingSpeedController(speeds, initialPosition, endBrakingPosition);
-            }
-            currentSpeedControllers.add(brakingSpeedController);
-            res.add(brakingSpeedController);
+        // coasting phase
+        var speed = initialSpeed;
+        var location = convertPosition(schedule, sim, initialPosition);
+        var coastingPhase = new SortedDoubleMap();
+        while (speed > targetSpeed && location.getPathPosition() < endPosition) {
+            var step = nextStep(
+                    location,
+                    speed,
+                    schedule.rollingStock,
+                    TIME_STEP,
+                    endPosition,
+                    1,
+                    (integrator) -> Action.coast());
+            coastingPhase.put(location.getPathPosition(), speed);
+            speed = step.finalSpeed;
+            location.updatePosition(schedule.rollingStock.length, step.positionDelta);
         }
-
-        // new running calculation with brakingSpeedController + scaled MapSpeedController
-        // now only the re-acceleration phase is missing
-        var newSpeeds = getExpectedSpeeds(sim, schedule, currentSpeedControllers, TIME_STEP,
-                initialPosition, endPosition, initialSpeed);
-
-        // backwards acceleration calculation
-        double speed = roiSpeeds.interpolate(endPosition); //The speed is calculated from the old running time
-        var location = convertPosition(schedule, sim, endPosition);
-        while (speed > newSpeeds.interpolate(location.getPathPosition())
-                && location.getPathPosition() - speed * TIME_STEP > 1e-5
-                && speed >= 1e-5) {
-            var directive = new SpeedDirective(newSpeeds.interpolate(location.getPathPosition()));
+        coastingPhase.put(location.getPathPosition(), targetSpeed);
+
+        // acceleration phase
+        speed = expectedSpeeds.interpolate(endPosition);
+        location = convertPosition(schedule, sim, endPosition);
+        var acceleratingPhase = new SortedDoubleMap();
```

Comment:
This variable is never queried

Reply:
I will call it with accelerationFirstPosition so that it is used, because i need to keep it for further modifications.